types/block.go

```diff
@@ -54,13 +54,47 @@ type Block struct {
 	// Proof is a proof of spacetime generated using the hash of the previous ticket as
 	// a challenge
 	Proof proofs.PoStProof `json:"proof"`
+
+	cachedCid cid.Cid
+
+	cachedBytes []byte
 }
 
+// set this to true to panic if the blocks data differs from the cached cid. This should
+// be obviated by changing the block to have protected construction, private fields, and
+// getters for all the values.
```

Comment:
can you file an issue for this against testnet and link to it here? hidden technical debt is what sunk the titanic.